src/libraries/System.Private.CoreLib/src/System/Net/WebUtility.cs

```diff
@@ -804,272 +804,290 @@ static HtmlEntities()
+                        0x65, 0x6C, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("le")*/      , 0x64, 0x22, /*'\x2264'*/
+                        0x65, 0x67, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("ge")*/      , 0x65, 0x22, /*'\x2265'*/
+                        0x62, 0x75, 0x73, 0x00, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("sub")*/     , 0x82, 0x22, /*'\x2282'*/
+                        0x70, 0x75, 0x73, 0x00, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("sup")*/     , 0x83, 0x22, /*'\x2283'*/
+                        0x62, 0x75, 0x73, 0x6E, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("nsub")*/    , 0x84, 0x22, /*'\x2284'*/
+                        0x65, 0x62, 0x75, 0x73, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("sube")*/    , 0x86, 0x22, /*'\x2286'*/
+                        0x65, 0x70, 0x75, 0x73, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("supe")*/    , 0x87, 0x22, /*'\x2287'*/
+                        0x73, 0x75, 0x6C, 0x70, 0x6F, 0x00, 0x00, 0x00 /*ToUInt64Key("oplus")*/   , 0x95, 0x22, /*'\x2295'*/
+                        0x73, 0x65, 0x6D, 0x69, 0x74, 0x6F, 0x00, 0x00 /*ToUInt64Key("otimes")*/  , 0x97, 0x22, /*'\x2297'*/
+                        0x70, 0x72, 0x65, 0x70, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("perp")*/    , 0xA5, 0x22, /*'\x22a5'*/
+                        0x74, 0x6F, 0x64, 0x73, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("sdot")*/    , 0xC5, 0x22, /*'\x22c5'*/
+                        0x6C, 0x69, 0x65, 0x63, 0x6C, 0x00, 0x00, 0x00 /*ToUInt64Key("lceil")*/   , 0x08, 0x23, /*'\x2308'*/
+                        0x6C, 0x69, 0x65, 0x63, 0x72, 0x00, 0x00, 0x00 /*ToUInt64Key("rceil")*/   , 0x09, 0x23, /*'\x2309'*/
+                        0x72, 0x6F, 0x6F, 0x6C, 0x66, 0x6C, 0x00, 0x00 /*ToUInt64Key("lfloor")*/  , 0x0A, 0x23, /*'\x230a'*/
+                        0x72, 0x6F, 0x6F, 0x6C, 0x66, 0x72, 0x00, 0x00 /*ToUInt64Key("rfloor")*/  , 0x0B, 0x23, /*'\x230b'*/
+                        0x67, 0x6E, 0x61, 0x6C, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("lang")*/    , 0x29, 0x23, /*'\x2329'*/
+                        0x67, 0x6E, 0x61, 0x72, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("rang")*/    , 0x2A, 0x23, /*'\x232a'*/
+                        0x7A, 0x6F, 0x6C, 0x00, 0x00, 0x00, 0x00, 0x00 /*ToUInt64Key("loz")*/     , 0xCA, 0x25, /*'\x25ca'*/
+                        0x73, 0x65, 0x64, 0x61, 0x70, 0x73, 0x00, 0x00 /*ToUInt64Key("spades")*/  , 0x60, 0x26, /*'\x2660'*/
+                        0x73, 0x62, 0x75, 0x6C, 0x63, 0x00, 0x00, 0x00 /*ToUInt64Key("clubs")*/   , 0x63, 0x26, /*'\x2663'*/
+                        0x73, 0x74, 0x72, 0x61, 0x65, 0x68, 0x00, 0x00 /*ToUInt64Key("hearts")*/  , 0x65, 0x26, /*'\x2665'*/
+                        0x73, 0x6D, 0x61, 0x69, 0x64, 0x00, 0x00, 0x00 /*ToUInt64Key("diams")*/   , 0x66, 0x26, /*'\x2666'*/
+                    };
+
+                var dictionary = new Dictionary<ulong, char>(tableData.Length / (sizeof(ulong) + sizeof(char)));
```

Comment:
I think Jan was suggesting not using Dictionary at all, and instead a perfect hash over a buffer.

If you use what you're doing here, looks like you can remove the byte in the 8th column as it's always zero.

Reply:
Well, this code is only executed once, my goal was to decrease the native size (from `17605` bytes to only `277` bytes!) using the first Jan's suggestion. If you insist on the second more performant one I can change!

Reply:
No I think this is fine, I was secretly hoping you would do it only because it would be interesting 😁

Reply:
And kudos for the huge improvemnet. Are there any more of these big static dictionaries I wonder?

Reply:
Found another, but likely not on a hot path:
https://github.com/dotnet/runtime/blob/0e8efce08c99fb4b3ee0d46839dbe4a05c996b84/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs#L274

Reply:
Will take a look, thanks!